core/domain/rights_manager.py

```diff
@@ -382,7 +385,37 @@ def create_new_collection_rights(collection_id, committer_id):
     subscription_services.subscribe_to_collection(committer_id, collection_id)
 
 
-def get_collection_rights(collection_id, strict=True):
+@overload
+def get_collection_rights(
+    collection_id: str
+) -> rights_domain.ActivityRights: ...
+
+
+@overload
+def get_collection_rights(
+    collection_id: str,
+    strict: Literal[True]
+) -> rights_domain.ActivityRights: ...
+
+
+@overload
+def get_collection_rights(
+    collection_id: str,
+    strict: Literal[False]
+) -> Optional[rights_domain.ActivityRights]: ...
+
+
+@overload
```

Comment:
Just curious about why you need both Literal and bool overloads here?

Reply:
When some call this method with `strict` as a keyword argument, like: 
```
def get_collection_rights('some_id', strict=False):...
``` 
Then, bool's overload is checked.

If someone calls this method with `strict` as a positional argument, like:
```
def get_collection_rights('some_id', False): ...
```
Then, Literal's overload is checked.

Reply:
Similar approach is used for other overloads.